src/openrct2/paint/Paint.cpp

```diff
@@ -306,42 +306,26 @@ static bool check_bounding_box(const paint_struct_bound_box& initialBBox, const
 
 template<> bool check_bounding_box<0>(const paint_struct_bound_box& initialBBox, const paint_struct_bound_box& currentBBox)
 {
-    if (initialBBox.z_end >= currentBBox.z && initialBBox.y_end >= currentBBox.y && initialBBox.x_end >= currentBBox.x
-        && !(initialBBox.z < currentBBox.z_end && initialBBox.y < currentBBox.y_end && initialBBox.x < currentBBox.x_end))
-    {
-        return true;
-    }
-    return false;
+    return initialBBox.z_end >= currentBBox.z && initialBBox.y_end >= currentBBox.y && initialBBox.x_end >= currentBBox.x
+        && !(initialBBox.z < currentBBox.z_end && initialBBox.y < currentBBox.y_end && initialBBox.x < currentBBox.x_end);
```

Comment:
All of these are slightly easier to read in their old notation (though still awful).

Reply:
Perhaps split the bounding box expressions into three: a separate check for X, Y, and Z. Then, in the end, return the conjunction of the three.

Reply:
Yeah, let's go with that.

Reply:
As discussed on gitter, I've reverted it for now.